track_and_trace/asset_client/utils/submit_utils.js

```diff
@@ -0,0 +1,338 @@
+  })
+  inputAddresses = inputAddresses.concat(inputs)
+
+  var outputAddresses = [
+    ...addressing.computeContractRegistryAddresses(contracts),
+    ...addressing.computeContractAddresses(contracts)
+  ]
+
+  outputs.forEach(function(output) {
+    outputAddresses.push(addressing.computeNamespaceRegistryAddress(output))
+  })
+  outputAddresses = outputAddresses.concat(outputs)
+
+  const transactionHeaderBytes = TransactionHeader.encode({
+    familyName: addressing.sabreFamily.name,
+    familyVersion: addressing.sabreFamily.version,
+    inputs: inputAddresses,
+    outputs: outputAddresses,
+    signerPublicKey: pubkey,
+    batcherPublicKey: pubkey,
+    dependencies: [],
+    payloadSha512: createHash('sha512').update(sabrePayloadBytes).digest('hex')
+  }).finish()
+
+  let signature = signer.sign(transactionHeaderBytes)
+
+  return Transaction.create({
+    header: transactionHeaderBytes,
+    headerSignature: signature,
+    payload: sabrePayloadBytes
+  })
+}
+
+const submitTxns = (transactions, signer) => {
+  let transactionIds = transactions.map((txn) => txn.headerSignature)
+  let signerPublicKey = signer.getPublicKey().asHex()
+
+  const batchHeaderBytes = BatchHeader.encode({
+    signerPublicKey,
+    transactionIds
+  }).finish()
+
+  let signature = signer.sign(batchHeaderBytes)
+
+  const batch = Batch.create({
+    header: batchHeaderBytes,
+    headerSignature: signature,
+    transactions
+  })
+
+  const batchListBytes = BatchList.encode({
+    batches: [batch]
+  }).finish()
+
+  return request({
+    method: 'POST',
+    url: `${GRID_SERVER}/batches`,
+    headers: { 'Content-Type': 'application/octet-stream' },
+    body: batchListBytes
+  })
+    .then((body) => {
+      let link = JSON.parse(body).link
+      _waitForCommit(
+          transactionIds,
+          _formStatusUrl(link)
+      )
+      return signer
+    })
+    .catch((err) => {
+      console.log(err)
+    })
+}
+
+const createSeedUser = (org, signer, encryptedPrivateKey) => {
+  let publicKey = signer.getPublicKey().asHex()
+
+  return request({
+    method: 'POST',
+    url: `${TNT_SERVER}/users`,
+    body: {
+      email: org.email,
+      password: org.password,
+      encryptedPrivateKey,
+      publicKey
+    },
+    json: true
+  })
+  .catch((e) => {
+    if (e.error && e.statusCode === 400) {
+      return Promise.reject(e.error.message)
+    } else {
+      return Promise.reject('Unable to create seed user')
+    }
+  })
+  .then((result) => {
+    if (result.status === 'ok') {
+      return {org, signer}
+    }
+  })
+}
+
+const createSchemas = (org, signer) => {
+  let txns = org.schemas.map((schema) => createSchemaTransaction(schema, signer))
+
+  return submitTxns(txns, signer)
+}
+
+const createSchemaTransaction = (schema, signer) => {
+  if (!signer) {
+    console.error('a signer must be provided')
+  }
+
+  let schemaCreate= protos.SchemaCreateAction.create({
```

Comment:
Looks like some strange formatting. I suggest running eslint on this file

Reply:
updated